warehouse/redshift.go

```diff
@@ -295,3 +325,56 @@ func (rs *Redshift) DoesTableExist(name string) bool {
 	}
 	return (exists != 0)
 }
+
+func (rs *Redshift) getTableColumns(name string) []string {
+	ctx := context.Background()
+	log.Printf("Fetching columns for table %s", name)
+	rows, err := rs.conn.QueryContext(ctx,
+		"SELECT \"column\" FROM pg_table_def WHERE schemaname = 'public' AND tablename = $1;", name)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var columns []string
+
+	defer rows.Close()
+	for rows.Next(){
+		var column string
+		err = rows.Scan(&column)
+		columns = append(columns, column)
+	}
+
+	// get any error encountered during iteration
+	if err = rows.Err(); err != nil {
+		log.Fatal(err)
+	}
+	return columns
+}
+
+func (rs *Redshift) getAlterTableStatement(f WarehouseField, tableName string) string {
+	return fmt.Sprintf("ALTER TABLE %s ADD COLUMN %s %s;", tableName, f.Name, f.DBType)
```

Comment:
one last nit: Now that this function is one line, it kind of doesn't need to be a function anymore? For the clean code purists this would be a positive for readability's sake, but I think most go coders would be unhappy with the unnecessary overhead of adding an additional frame to the stack.

Reply:
I actually have a strong preference for doing something like this. Readability is priceless! Happy to change though, if you think the downsides in GO are decent enough to warrant a non fn approach!

Reply:
a weekend later, and after reading some of the comments in the slack channel, since my 'strong' preference is wavering, changing to inline!